### PR TITLE
Basic classes

### PR DESCRIPTION
**Pull request by preciousuwhubetine**
In this branch, I created three classes:
1. A `person` class.
2. A `student` class (inherits from `person`).
3. A `teacher` class (inherits from `person`).

### CODE REVIEW

File: classes/person.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
+class Person
+  attr_accessor :name, :age
+  attr_reader :id
+
+  def initialize(age, name = 'Unknown', parent_permission: true)
+    @id = nil
```
Comment (Ismailco) [inline classes/person.rb]: - [ ] Kindly make sure the id is a number generated randomly, you can use `Random.ran` to generate a random id for the person

File: classes/person.rb [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+  def of_age?
+    @age >= 18
+  end
+
+  public
+
+  def can_use_services?
+    of_age? && @parent_permission
+  end
```
Comment (Ismailco) [inline classes/person.rb]: - [ ] [OPTIONAL] You can start with this method `can_use_services?` and you don't have to add a public flag

**Review by Ismailco — CHANGES_REQUESTED** [review]
### Hi @preciousbetine, 

Good job so far 👍🏼 
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

✔️ No linter errors
✔️ Used Gitflow
✔️ Work documented professionally

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: classes/person.rb [hunk 3]
```diff
@@ -0,0 +1,21 @@
+class Person
+  attr_accessor :name, :age
+  attr_reader :id
+
+  def initialize(age, name = 'Unknown', parent_permission: true)
+    @id = rand
```
Comment (amejid) [inline classes/person.rb:6]: - [ ] [OPTIONAL] `rand` function generates a random number between 0 and 1, usually a long decimal number. Please specify a range to generate a random number from, that way it'll make it easier to use it in the future. 😎️
**Example**
`rand(1..1000)`

**Review by amejid — APPROVED** [review]
Hi @preciousbetine, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights
- Linter check passing :heavy_check_mark: 
- Correct workflow :heavy_check_mark: 
- Classes and methods created :heavy_check_mark: 
- Inheritance utilized :heavy_check_mark: 
- Changes implemented :heavy_check_mark: 
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**